File: benchmarks/pandas/bench_keep_true_false.py
"""Benchmark: keepTrue / keepFalse equivalent — boolean mask filtering on a 100k-element Series"""
import json
import time
import pandas as pd
import numpy as np

N = 100_000
WARMUP = 2
ITERATIONS = 5

data = list(range(N))
mask = [i % 2 == 0 for i in range(N)]
s = pd.Series(data, dtype=float)
bool_mask = pd.array(mask, dtype=bool)

for _ in range(WARMUP):
    s[bool_mask]
    s[~bool_mask]

start = time.perf_counter()
for _ in range(ITERATIONS):
    s[bool_mask]
    s[~bool_mask]
total = (time.perf_counter() - start) * 1000

print(json.dumps({
    "function": "keep_true_false",
    "mean_ms": total / ITERATIONS,
    "iterations": ITERATIONS,
    "total_ms": total,
}))

File: benchmarks/tsb/bench_keep_true_false.ts
/**
 * Benchmark: keepTrue / keepFalse — boolean mask filtering on a 100k-element Series
 */
import { Series, keepTrue, keepFalse } from "../../src/index.js";

const N = 100_000;
const WARMUP = 2;
const ITERATIONS = 5;

const data = Array.from({ length: N }, (_, i) => i * 1.0);
const mask = Array.from({ length: N }, (_, i) => i % 2 === 0);
const s = new Series({ data });

for (let i = 0; i < WARMUP; i++) {
  keepTrue(s, mask);
  keepFalse(s, mask);
}

const start = performance.now();
for (let i = 0; i < ITERATIONS; i++) {
  keepTrue(s, mask);
  keepFalse(s, mask);
}
const total = performance.now() - start;

console.log(
  JSON.stringify({
    function: "keep_true_false",
    mean_ms: total / ITERATIONS,
    iterations: ITERATIONS,
    total_ms: total,
  }),
);
